xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-handlers/xwiki-platform-extension-handler-xar/src/main/java/org/xwiki/extension/xar/internal/security/XarSecurityRule.java

```diff
@@ -78,4 +79,24 @@ public RuleState getState()
     {
         return RuleState.DENY;
     }
+
+    @Override
+    public int hashCode()
+    {
+        return new HashCodeBuilder(3, 17).append(right).append(simple)
+                .append(securityTool).toHashCode();
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (o instanceof XarSecurityRule) {
+            XarSecurityRule rule = (XarSecurityRule) o;
+
+            return this.simple == rule.simple && this.right == rule.right
```

Comment:
I guess you could use the `EqualsBuilder` too there: https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/#HEquals2FHashCodeandToStringimplementations

Reply:
Nice, thanks for the tip ; updating …

Reply:
Fixed in 3bddb86